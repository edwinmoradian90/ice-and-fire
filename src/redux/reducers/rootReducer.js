import { combineReducers } from 'redux'
import mainReducer from '../reducers/mainReducer'
import categoriesReducer from '../reducers/categoriesReducer'

export default combineReducers({
    mainReducer,
    categoriesReducer
})