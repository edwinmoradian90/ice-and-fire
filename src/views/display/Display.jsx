import React, { useEffect } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import {
    FETCH_DISPLAY_REQUEST,
    FETCH_DISPLAY_SUCCESS,
    FETCH_DISPLAY_FAILURE
} from '../../redux/constants/constants'
import Header from '../../components/header/Header'
import Books from '../../components/list/displayListItems/Books'
import Characters from '../../components/list/displayListItems/Characters'
import Houses from '../../components/list/displayListItems/Houses'
import axios from 'axios'
import { url } from '../../config/config'

function Display(props) {
    const dispatch = useDispatch()
    const loading = useSelector(state => state.displayReducer.loading)
    const items = useSelector(state => state.displayReducer.data) || ''
    const currentUrl = props.location.pathname.split(('/'))
    const apiUrl = `${url}/${currentUrl[1]}/${currentUrl[2]}`

    console.log(currentUrl)

    useEffect(() => {
        dispatch({
            type: FETCH_DISPLAY_REQUEST,
            payload: {
                loading: true,
            }
        })
        axios.get(apiUrl)
            .then(res => {
                dispatch({
                    type: FETCH_DISPLAY_SUCCESS,
                    payload: {
                        loading: false,
                        data: res,
                    }
                })
            })
            .catch(err => {
                dispatch({
                    type: FETCH_DISPLAY_FAILURE,
                    payload: {
                        loading: false,
                        error: err
                    }
                })
            })
    }, [])
    console.log(currentUrl[1])

    return (
        <>
            {
                !loading ?
                    (
                        <div>
                            <Header name={'something'} />
                            {
                                currentUrl[1] === 'books'
                                    ? <Books items={items} />
                                    : null
                            }
                            {
                                currentUrl[1] === 'characters'
                                    ? <Characters items={items} />
                                    : null
                            }
                            {
                                currentUrl[1] === 'houses'
                                    ? <Houses items={items} />
                                    : null
                            }
                        </div>
                    )
                    :
                    (
                        <p>loading...</p>
                    )
            }
        </>
    )
}

export default Display