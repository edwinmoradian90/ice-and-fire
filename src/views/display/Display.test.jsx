import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Display from './Display';
import rootReducer from '../../redux/reducers/rootReducer';

describe('Display', () => {
    const store = createStore(rootReducer);
    const location = {
        pathname: 'https://anapioficeandfire/api/characters/200'
    }
    it('Matches snapshot', () => {
        const component = renderer.create(
            <Provider store={store}>
                <Router>
                    <Display location={location} />
                </Router>
            </Provider>
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});