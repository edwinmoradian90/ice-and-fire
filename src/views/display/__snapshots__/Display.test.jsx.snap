// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Matches snapshot 1`] = `
<div>
  <div
    className="header"
  >
    <a
      className="sc-AxjAm irMvbk"
      href="/"
      onClick={[Function]}
    >
      <h1
        className="sc-AxhCb kYEhcE appTitle"
      >
        Ice and Fire
      </h1>
    </a>
  </div>
</div>
`;
